SoftLayer/order_utils.py

```diff
@@ -0,0 +1,145 @@
+"""Utility functions for mapping packages and categories to complex types."""
+
+
+def get_package_type_map():
+    """Gets a map of package->complexType."""
+    return {
+        'ADDITIONAL_SERVICES_APPLICATION_DELIVERY_APPLIANCE':
+            'SoftLayer_Container_Product_Order_Network_Application_Delivery_Controller',
+        'BARE_METAL_GATEWAY':
+            'SoftLayer_Container_Product_Order_Hardware_Server_Gateway_Appliance',
+        'GATEWAY_RESOURCE_GROUP':
+            'SoftLayer_Container_Product_Order_Gateway_Appliance_Cluster',
+        'HARDWARE_SECURITY_MODULE':
+            'SoftLayer_Container_Product_Order_Hardware_Security_Module',
+    }
+
+
+def get_category_type_map():
```

Comment:
I'm not sure how I feel about this level of hard coding categories. 
Having the user figure out the complexType on their own might be preferable here. Thoughts?

Reply:
Yeah, my goal was to just make this as simple as possible for the user, because it's not obviously clear where to find the SoftLayer_Container_Product_Order type that needs to be used. Unfortunately, this was the only way to do that (there is no API call to properly get the type).

Maybe I should just move this out into a different PR that we can debate on (it'll thankfully make this change a little bit smaller). I'll make complexType a kwarg and CLI option so one day we'll be able to make it optional, but for now I'll require it to be set. How does that sound?

Reply:
Yeah thats a good plan.